src/Image.php

```diff
@@ -88,12 +88,14 @@ public function lazyloadBackgroundImages( $html, $buffer ) {
 	 * @return string
 	 */
 	private function addLazyClass( $element ) {
-		if ( preg_match( '#class=["\']?(?<classes>[^"\'>]*)["\']?#is', $element, $class ) ) {
+		if ( preg_match( '#class\s*=\s*(?<classes>["\'].*?["\']|[^\s]+)#is', $element, $class ) ) {
+			$contains_quotes  = preg_match( '#^(\'|").+(\'|")$#', $class['classes'] );
+			$class['classes'] = str_replace( [ '"', '\'' ], '', $class['classes'] );
 			if ( empty( $class['classes'] ) ) {
 				return str_replace( $class[0], 'class="rocket-lazyload"', $element );
 			}
 
-			$classes = str_replace( $class['classes'], $class['classes'] . ' rocket-lazyload', $class[0] );
+			$classes = str_replace( $class['classes'], ( ! $contains_quotes ? '"' : '' ) . $class['classes'] . ' rocket-lazyload' . ( ! $contains_quotes ? '"' : '' ), $class[0] );
```

Comment:
For this one, the default of the ternaries concatenates an empty `''` string. That's unnecessary and adds processing time. 

Instead, create a variable with the replacement class:

```suggestion
			$contains_quotes  = preg_match( '#^(\'|").+(\'|")$#', $class['classes'] );
			$replacement_class = $class['classes'] . ' rocket-lazyload';
			// Does not contain quotes.
			if ( ! preg_match( '#^(\'|").+(\'|")$#', $class['classes'] ) ) {
				$replacement_class = '"' . $replacement_class . '"';
			}
			$classes = str_replace( $class['classes'], $replacement_class, $class[0] );
```